### DISCUSSION TITLE
Check if a step is completed

### DISCUSSION THREAD

**Discussion by fabio-algomera** [spatie/laravel-livewire-wizard#67] [Q&A]
Hello everyone,
do you think there is a way to tell if a step is completed?
I am using this wizard to create a model and later edit it. Upon editing, however, I would like to see if a step has been completed (so all the required fields in the step have been filled in).
Checking the documentation, the only available methods for $step are: isPrevious(), isNext(), isCurrent(). is there any way to create an isCompleted() method indicating the completion of the required fields?

I have tried indicating "completed" as the initial state but mostly during navigation this data is lost.

Thank you very much to those who can answer me!

**Comment by ju5t** [comment]
You would have to implement this yourself. It's impossible for this package to decide when a step is completed or not. 

This page might help you: https://spatie.be/docs/laravel-livewire-wizard/v1/usage/accessing-state#content-using-a-custom-state-object. If you change it to a `isCompleted` method with the logic you need you're pretty much there.

**Reply by fabio-algomera** [reply]
Thank you for the answer.
I'll try it. :)

**Comment by alexfaus08** [comment]
@fabio-alkimia did you ever implement this? I am trying to do something similar currently.
